src/backend/access/transam/xact.c

```diff
@@ -2317,14 +2317,14 @@ CommitTransaction(void)
 	willHaveObjectsFromSmgr =
 			PersistentEndXactRec_WillHaveObjectsFromSmgr(EndXactRecKind_Commit);
 
-	if (willHaveObjectsFromSmgr)
-	{
-		/*
-		 * We need to ensure the recording of the [distributed-]commit record and the
-		 * persistent post-commit work will be done either before or after a checkpoint.
-		 */
-		CHECKPOINT_START_LOCK;
-	}
```

Comment:
I have some concern about removing this lock. As the comment indicated, what will happen in such case?

1) flush commit record.
2) start a checkpoint.
3) persistent post-commit work (change persistent table)